Zend/zend.h

```diff
@@ -339,7 +339,21 @@ extern ZEND_API size_t (*zend_printf)(const char *format, ...) ZEND_ATTRIBUTE_PT
 extern ZEND_API zend_write_func_t zend_write;
 extern ZEND_API FILE *(*zend_fopen)(zend_string *filename, zend_string **opened_path);
 extern ZEND_API void (*zend_ticks_function)(int ticks);
+
+/* Called by the VM in certain places like at the end of function calls or
+ * jumps, if EG(vm_interrupt) has been set.
+ *
+ * If this is used to switch the EG(current_execute_data), such as implementing
+ * a coroutine scheduler, then it needs to check the top frame to see if it's
+ * an internal function. If so, it needs to do something similar to a fiber
+ * switch, or at the very least delay switching until the internal frame is no
+ * longer on top. Prior to PHP 8.0, this check was not necessary. In PHP 8.0
+ * zend_call_function started calling zend_interrupt_function, and in 8.4 the
+ * DO_*CALL* opcodes started calling the zend_interrupt_function while the
+ * internal frame is still on top.
+ */
```

Comment:
@dstogov And also check the wording here?

Reply:
``zend_interrupt_function`` may be called by VM if ``EG(vm_interrupt)`` flag was asynchronously set. VM checks for ``EG(vm_interrupts)`` at PHP loop header (to break long-running loops), at PHP user function entry (to break recursion), and after internal function calls (that may take long time).

I would also remove the fiber related suggestion. At least I don't know how to do this...